src/fast_sincos.c

```diff
@@ -0,0 +1,407 @@
+#include <stdio.h>
+#include <math.h>
+#include "fast_sincos.h"
+#include "utils.h"
+
+static fast_sincos_real fastSinChebyshev(const fast_sincos_real angleRadians, const int sinDegree);
+static fast_sincos_real fastCosChebyshev(const fast_sincos_real angleRadians, const int sinDegree);
+static fast_sincos_real lookupCosInterpolate(const fast_sincos_real angleRadians);
+static fast_sincos_real lookupSinInterpolate(const fast_sincos_real angleRadians);
+static fast_sincos_real lookupCos(const fast_sincos_real angleRadians);
+static fast_sincos_real lookupSin(const fast_sincos_real angleRadians);
+static inline fast_sincos_real scaleValueToRadians(const fast_sincos_real value);
+
+#ifdef USE_DEFAULT_SINE_TABLE
```

Comment:
I think it would be easier to use it we had an #ifndef here rather than an #ifdef, as a user, I don't want to have to export some shady variables to use fast sin / fast cos, I just want to include the header file and use it.